### ISSUE TITLE
Tree behavior doesn't calculate lft/rght correctly on creation

### ISSUE THREAD

**Issue by sdustinh** [cakephp/cakephp#8264]
All of this is done with classes generated by `bake`. I thought may be it was something I had configured incorrectly but I have been able to replicate it with straight generated classes as well. Here is the setup...

```sql
CREATE TABLE IF NOT EXISTS `device_type_groups` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `parent_id` int(11) DEFAULT NULL,
  `lft` int(11) DEFAULT NULL,
  `rght` int(11) DEFAULT NULL,
  `created` datetime DEFAULT NULL,
  `modified` datetime DEFAULT NULL,
  `name` varchar(255) NOT NULL,
  PRIMARY KEY (`id`),
  KEY `parent_id` (`parent_id`,`lft`,`rght`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 AUTO_INCREMENT=1 ;
```

```php
// DeviceTypeGroupsTable.php
public function initialize(array $config)
{
    parent::initialize($config);
    
    $this->table('device_type_groups');
    $this->displayField('name');
    $this->primaryKey('id');
    
    $this->addBehavior('Timestamp');
    $this->addBehavior('Tree');
    
    $this->hasMany('ChildGroups', [
        'className' => 'DeviceTypeGroups',
        'dependent' => true,
        'foreignKey' => 'parent_id'
    ]);
    $this->belongsTo('ParentGroups', [
        'className' => 'DeviceTypeGroups',
        'foreignKey' => 'parent_id'
    ]);
}
```

```php
// Adding data...
$groups = [
    ['name' => 'Environmental'],
    ['name' => 'Network'],
    ['name' => 'Power'],
    ['name' => 'Server'],
    [
        'name' => 'Switch',
        'child_groups' => [
            ['name' => 'Access'],
            ['name' => 'Core'],
            ['name' => 'Layer 3'],
        ]
    ],
    ['name' => 'Uplink']
];

foreach ($groups as $data) {
    $entity = $this->DeviceTypeGroups->newEntity($data);
    $this->DeviceTypeGroups->save($entity);
}
```

```sql
// Resulting records...
INSERT INTO `device_type_groups` (`id`, `parent_id`, `lft`, `rght`, `created`, `modified`, `name`) VALUES
(1, NULL, 1, 2, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Environmental'),
(2, NULL, 3, 4, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Network'),
(3, NULL, 5, 6, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Power'),
(4, NULL, 7, 8, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Server'),
(5, NULL, 9, 16, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Switch'),
(6, 5, 10, 11, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Access'),
(7, 5, 12, 13, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Core'),
(8, 5, 14, 15, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Layer 3'),
(9, NULL, 16, 17, '2016-02-15 21:17:59', '2016-02-15 21:17:59', 'Uplink');
```

There is a key overlap with the "Switch" and "Uplink"  records. I have also tried to fix this by using a transaction but it does the same thing. Running `recover()` does fix it.

CakePHP: 3.2.1
MySQL: 5.6.28
PHP: 5.6.18
OS: Ubuntu 14.04

**Comment by lorenzo** [comment]
I believe this is an issue recently fixed in the master branch. Could you please verify?

**Comment by sdustinh** [comment]
It looks like this was fixed in #8245 and I just didn't find an issue for it. I pulled down 08e43f7cb8cd4ea37bf380da6b704c206fead6f0 to test it out and it seems to be fixed. Sorry for not pulling master first. I guess I just wasn't even paying attention. :smirk:

**Comment by lorenzo** [comment]
Thanks for checking, and sorry for breaking the behavior :)
